packages/server/src/scim.ts

```diff
@@ -699,6 +703,96 @@ export class ScimServer implements DirectoryProvider {
+
+            // TODO: Remove this
+            console.log(JSON.stringify({ xmlSamlResponse }, null, 2));
+
+            // NOTE: While this isn't super reliable, it makes it unnecessary to require _heavy_ (and generally security-vulnerability-filled) XML-parsing dependencies.
+            const anyPropertyEnd = "</saml2:AttributeValue>";
+            const emailPropertyStart = `<saml2:AttributeStatement><saml2:Attribute Name="email"><saml2:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:anyType">`;
+            const firstNamePropertyStart = `<saml2:Attribute Name="firstName"><saml2:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:anyType">`;
+            const lastNamePropertyStart = `<saml2:Attribute Name="lastName"><saml2:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:anyType">`;
+
+            const emailPropertyStartIndex = xmlSamlResponse.indexOf(emailPropertyStart);
+            const firstNamePropertyStartIndex = xmlSamlResponse.indexOf(firstNamePropertyStart);
+            const lastNamePropertyStartIndex = xmlSamlResponse.indexOf(lastNamePropertyStart);
+
+            const emailPropertyEndIndex =
+                emailPropertyStartIndex !== -1 ? xmlSamlResponse.indexOf(anyPropertyEnd, emailPropertyStartIndex) : -1;
+            const firstNamePropertyEndIndex =
+                firstNamePropertyStartIndex !== -1
+                    ? xmlSamlResponse.indexOf(anyPropertyEnd, firstNamePropertyStartIndex)
+                    : -1;
+            const lastNamePropertyEndIndex =
+                lastNamePropertyStartIndex !== -1
+                    ? xmlSamlResponse.indexOf(anyPropertyEnd, lastNamePropertyStartIndex)
+                    : -1;
+
+            const email =
+                emailPropertyStartIndex !== -1 && emailPropertyEndIndex !== -1
+                    ? xmlSamlResponse.substring(
+                          emailPropertyStartIndex + emailPropertyStart.length,
+                          emailPropertyEndIndex
+                      )
+                    : "";
+            const firstName =
+                firstNamePropertyStartIndex !== -1 && firstNamePropertyEndIndex !== -1
+                    ? xmlSamlResponse.substring(
+                          firstNamePropertyStartIndex + firstNamePropertyStart.length,
+                          firstNamePropertyEndIndex
+                      )
+                    : "";
+            const lastName =
+                lastNamePropertyStartIndex !== -1 && lastNamePropertyEndIndex !== -1
+                    ? xmlSamlResponse.substring(
+                          lastNamePropertyStartIndex + lastNamePropertyStart.length,
+                          lastNamePropertyEndIndex
+                      )
+                    : "";
```

Comment:
@MaKleSoft your thoughts on this "mess" are welcome. I think it's still better than adding an xml parser dependency.